MediaProcessor/src/DeepFilterCommandBuilder.cpp

```diff
@@ -0,0 +1,43 @@
+#include "DeepFilterCommandBuilder.h"
+#include <stdexcept>
+#include <sstream>
+
+namespace MediaProcessor {
+
+DeepFilterCommandBuilder& DeepFilterCommandBuilder::setInputAudio(const std::string& inputAudioPath) {
+    m_inputAudioPath = inputAudioPath;
+    addFlag("--input");
+    addArgument(inputAudioPath);
+    return *this;
+}
+
+DeepFilterCommandBuilder& DeepFilterCommandBuilder::setOutputAudio(const std::string& outputAudioPath) {
+    m_outputAudioPath = outputAudioPath;
+    addFlag("--output");
+    addArgument(outputAudioPath);
+    return *this;
+}
+
+DeepFilterCommandBuilder& DeepFilterCommandBuilder::setNoiseReductionLevel(double level) {
+    if (level < 0.0 || level > 1.0) {
+        throw std::invalid_argument("Noise reduction level must be between 0.0 and 1.0");
+    }
+    m_noiseReductionLevel = level;
+
+    addFlag("--noise-reduction");
```

Comment:
Here and in the line below, please use the correct overload of `addFlag` that allows passing a flag with a value. This will improve consistency. Could you update this and other similar occurrences?

Reply:
Will do, going to push this update right away!